File: ios/Classes/Bridge/GetABTestingVariantBridge.swift
import Usercentrics
import Flutter

struct GetABTestingVariantBridge : MethodBridge {
    let name: String = "getABTestingVariant"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        result(usercentrics.shared.getABTestingVariant())
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetABTestingVariantBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class GetABTestingVariantBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
): MethodBridge {
    override val name: String
        get() = "getABTestingVariant"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        result.success(usercentrics.instance.getABTestingVariant())
    }
}
